src/CarManager.java

```diff
@@ -0,0 +1,135 @@
+import java.util.*;
+
+public class CarManager {
+    List<Car> carList = new ArrayList<>();
+    private Scanner scan = new Scanner(System.in);
+
+    public void searchByDiameter() {
+        System.out.println("Input diameter");
+        int inputDiameter = scan.nextInt();
+        for (int i = 0; i < carList.size(); i++) {
+            if (carList.get(i).getWheels().getDiameter() == inputDiameter) {
+                System.out.println(carList.get(i).getName());
+            }
+        }
+    }
+
+    public void searchByDiameterAndColor() {
+        System.out.println("Input diameter:");
+        int inputDiameter = scan.nextInt();
+        System.out.println("Input color:");
+        Colors inputColor = Colors.valueOf(scan.next().toUpperCase());
+        for (int i = 0; i < carList.size(); i++) {
+            if (carList.get(i).getWheels().getDiameter() == inputDiameter && carList.get(i).getBody().getColor() == inputColor) {
+                System.out.println(carList.get(i).getName());
+            }
+        }
+    }
+
+    public void replaceSteeringWheelsOfRedCars() {
+        for (int i = 0; i < carList.size(); i++) {
+            if (carList.get(i).getBody().getColor() == Colors.RED) {
+                String preferedSteeringWheelType = scan.next();
+                if (preferedSteeringWheelType == "Multifunctioned") {
+                    carList.get(i).getSteeringWheel().setType(true);
+                } else if (preferedSteeringWheelType == "Common") {
+                    carList.get(i).getSteeringWheel().setType(false);
+                    ;
+                } else {
+                    System.out.println("You have chosen wrong steering wheel type. Failed changing");
+                }
+            }
+        }
+    }
+
+    public void multiplyDiameterByTwoWhenSteeringWheelIsMultifunctioned() {
+        for (int i = 0; i < carList.size(); i++) {
+            if (carList.get(i).getSteeringWheel().isMuiltifunctioned()) {
+                carList.get(i).getWheels().setDiameter(carList.get(i).getWheels().getDiameter() * 2);
+            }
+        }
+    }
+
+    public void replaceCarsWhenDiameterIsLessThanEntered() {
+        System.out.println("Input diameter:");
+        int inputDiameter = scan.nextInt();
+        for (int i = 0; i < carList.size(); i++) {
+            if (carList.get(i).getWheels().getDiameter() < inputDiameter) {
+                carList.add(i, new Car());
+            }
+        }
+    }
+
+    public void findCarsWithInputBodyType() {
+        ListIterator<Car> listIterator = carList.listIterator();
+        System.out.println("Input body type:");
+        BodyTypes inputGBodyType = BodyTypes.valueOf(scan.next().toUpperCase());
+        while (listIterator.hasNext()) {
+            listIterator.next();
+            boolean inputBodyTypeIsEqualToInitial = carList.get(listIterator.previousIndex()).getBody().getType().equals(inputGBodyType);
```

Comment:
Repeat Iterator.

What value is returned by 'listIterator.next();'? Can it be used instead of 'carList.get(listIterator.previousIndex())'?
Should Iterator or ListIterator be used here?
Why not .getType() == inputGBodyType?

Reply:
Got you.